documentation/add/09_Serializable_Interfaces.md

```diff
@@ -0,0 +1,132 @@
+
+**Table 1: Main abstact methods requesing overloading (definition)**
+
+| Method | Description |
+|-----|---------|
+|  classVersion | returns the number, which describes current class version. E.g. 1 - for version 1|
+| saveableFields | returns list of property names, defining the class public interface |
+
+As soon as these properties are defined, one can use other class methods to serialize/deserialize objects and convert objects to/from structure. The main features of the available methods are summarized in the Tables 2:
+** Table 2 Main methods, available for serializable objects:**
+
+| Method | Description |
+|-----|---------|
+|  `struct = to_bare_struct(obj)` | returns structure, with the field names equal to the names returned by `saveableFields` method and values, defined by the values of these fields. If the values of fields are the serializable objects themselves, they are converted into correspondent structures recursively |
+| `obj = from_bare_struct(obj,data)` | Reverse of `to_bare_struct` method. Sets the values of the public properties of the serializable interface from the structure, obtained from `to_bare_struct` method |
+| `data = to_struct(obj)` | converts `obj` or array of `obj` into the `data` structure using `to_bare_struct` method and adds complimentary information about the class name and the size and shape of the `obj` array if relevant. |
+| *Static:* | |
+| `obj = from_struct(data)` | Reverse of `to_struct` method, recovering the object or array of objects from the `data`, generated by the `to_struct` method |
+
+Standard `loadobj/saveobj` and `serialize/deserialize` methods use the data obtained from the methods described in **Table 2** or the methods themselves to provide/use input for/from Matlab save/load methods or serialize/deserialize methods to save/load data to/from `.mat` files or transform the objects from/to linear arrays of bytes for communications over serial pipes. The `serialize/deserialize` routines use Herbert-defined `serialize/deserialize` methods over the structures obtained using `to_struct/from_struct` methods.
+
+Two remaining methods presented on the **Fig 1** namely static `loadobj` and `from_old_struct` methods need overloading if/when you want to support loading the versions of a class, different from the the current class version. By default, a standard `loadobj` method exists and this method expects to receive the structure, produced by `to_struct` method. `to_struct` method adds information about the class name, so generic `loadobj` extracts this information and recovers the class. If this information is missing, which may happen when you loading a class structure, generated before the object become `serializable`, you need to overload `loadobj` to provide the empty instance of the class, as described by the code snippet below:
```

Comment:
versions of a class, different from the current class version
(1) remove comma after class
(2) different from -> earlier than
New paragraph after "from the current class version2.

So you're saying that if I save an object which is already serializable, then the custom loadobj here is not necessary to retrieve it. It could be worth saying how the code knows that this is the case and does  not call the custom loadobj.

Reply:
2 -- different -- comparison goes on equality and its theoretically possible to have version higher then current. You may want to downgrade

Reply:
How about that?